packages/appsync-emulator-serverless/lambdaRunner.js

```diff
@@ -21,15 +29,20 @@ process.once('message', ({ module, handlerPath, handlerMethod, payload }) => {
 
     log.info('invoke', handlerMethod);
     const context = {};
-    handlerModule[handlerMethod](payload, context, (err, output) => {
-      if (err) {
-        sendErr(err);
-        return;
-      }
-      process.send({ type: 'success', output }, () => {
-        process.exit();
+
+    Promise.resolve(handlerModule[handlerMethod](payload, context, onResponse))
```

Comment:
Instead of always wrapping in a `Promise.resolve` let's check to see if the method returns something "thenable".  For example:

```
const methodResult = handlerModule[handlerMethod](payload, context, onResponse)

if(methodResult.then) {
  try {
    const output = await methodResult
    process.send({ type: 'success', output }, process.exit);
  } catch(e) {
    sendErr(e)
  }
}
```

Reply:
Also, please add a unit test case for a lambda method that returns a promise.